src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DoNotIgnoreReturnValueCSharpTests.cs

```diff
@@ -0,0 +1,467 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.NetCore.Analyzers.Runtime;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.DoNotIgnoreReturnValueAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.CodeAnalysis.NetAnalyzers.UnitTests.Microsoft.NetCore.Analyzers.Runtime
+{
+    public class DoNotIgnoreReturnValueCSharpTests
+    {
+        private readonly DiagnosticDescriptor doNotIgnoreRule = DoNotIgnoreReturnValueAnalyzer.DoNotIgnoreReturnValueRule;
+        private readonly DiagnosticDescriptor doNotIgnoreRuleWithMessage = DoNotIgnoreReturnValueAnalyzer.DoNotIgnoreReturnValueRuleWithMessage;
+
+        private const string attributeImplementationCSharp = $$"""
+            namespace System.Diagnostics.CodeAnalysis
+            {
+                [System.AttributeUsage(System.AttributeTargets.ReturnValue, AllowMultiple = false, Inherited = false)]
+                internal class DoNotIgnoreAttribute : System.Attribute
```

Comment:
Just curious when we are going to add the attribute in runtime and apply it to methods? It would be helpful to test the analyzer with real usages